### PR TITLE
Fix ts issues from cost

### LINKED ISSUE #1380: NotAuthorized.js
file: https://github.com/RedHatInsights/frontend-components/tree/master/packages/components/src/NotAuthorized/NotAuthorized.js

### LINKED ISSUE #1381: Maintenance.js
file: https://github.com/RedHatInsights/frontend-components/tree/master/packages/components/src/Maintenance/Maintenance.js

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://issues.redhat.com/browse/RHCLOUD-17757

closes #1380 
closes #1381 

### Changes
- fixed missing `children` prop from `Main.tsx`
- remove required `headingLevel`prop from `PageHeaderTitle` component
- Migrate Maintenance and NotAuthorized components to TS

### CODE REVIEW

File: packages/components/src/Maintenance/Maintenance.tsx [hunk 1]
```diff
@@ -1,11 +1,31 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import { EmptyState, EmptyStateBody, Title, Stack, StackItem, EmptyStateIcon } from '@patternfly/react-core';
+import { EmptyState, EmptyStateBody, Title, Stack, StackItem, EmptyStateIcon, EmptyStateProps } from '@patternfly/react-core';
 import { HourglassHalfIcon } from '@patternfly/react-icons';
 import './maintenance.scss';
 
-const Maintenance = ({ title, utcStartTime, utcEndTime, startTime, endTime, timeZone, description, redirectLink, className, ...props }) => {
+export interface MaintenanceProps extends Omit<Omit<EmptyStateProps, 'children'>, 'title'> {
```
Comment (rvsia) [inline packages/components/src/Maintenance/Maintenance.tsx:7]: `Omit<Test, "a"|"b">;` this does not work?

File: packages/components/src/PageHeader/PageHeaderTitle.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import classNames from 'classnames';
 import { Title, TitleProps } from '@patternfly/react-core';
 
-export interface PageHeaderTitleProps extends Omit<TitleProps, 'title'> {
+export interface PageHeaderTitleProps extends Omit<Omit<TitleProps, 'title'>, 'headingLevel'> {
```
Comment (rvsia) [inline packages/components/src/PageHeader/PageHeaderTitle.tsx:5]: 👁️

**Review by rvsia — APPROVED** [review]

File: packages/components/src/Maintenance/Maintenance.tsx [hunk 3]
```diff
@@ -33,26 +53,6 @@ const Maintenance = ({ title, utcStartTime, utcEndTime, startTime, endTime, time
...
-  startTime: '6am',
-  endTime: '8am',
-  timeZone: 'EST',
-  redirectLink: 'https://status.redhat.com/incidents',
-  title: 'Maintenance in progress',
-};
-
 export default Maintenance;
+
+<Maintenance title="x"></Maintenance>
```
Comment (rvsia) [inline packages/components/src/Maintenance/Maintenance.tsx:58]: Oh...

**Review by fhlavac — APPROVED** [review]
Tested it locally and looks good to me!

**Review by fhlavac — APPROVED** [review]

**Pull request merged**